client/src/BitsContainer.tsx

```diff
@@ -1,19 +1,26 @@
-import { Map } from 'immutable';
+import * as Immutable from 'immutable';
 import * as React from 'react';
 import { connect } from 'react-redux';
 import * as actionCreators from './action_creators';
 import Bit from './Bit';
 import { filterBits } from './bits_util';
+import { Bit as BitType } from './types';
 
-const BitsContainer = (props: any) => (
+type Props = {
+  bits?: Immutable.Map<string, BitType>
```

Comment:
Why are these nullable?

Reply:
IDK really how to handle this right now. This represents the properties on the object. And it maps back to the properties you put on the actual component. So if you have

```tsx
type Props = {
  prop1: string
  prop2: string
  prop3: string
}

<BitsContainer prop1={A} prop2={B}>
```

it will fail to compile because prop3 is not an optional field. I'm going to come back and see how to fix these Props objects in a future PR. this one is getting kind of big..